analytics/adobe-analytics/experience-accelerator/src/Client.ts

```diff
@@ -27,8 +31,11 @@ export abstract class Client {
         let contextKey = this.getContextKey(type);
         let candidates = this.getEvolv().context.get(contextKey) || [];
         for (let i = 0; i < candidates.length; i++) {
-            const allocation = this.lookupFromAllocations(candidates[i].cid);
-            this.sendMetrics(type, allocation);
+            if (!this.activeCandidateEvents[type][candidates[i].cid]) {
```

Comment:
Since this project is using TypeScript, for safety, I would use optional chaining
```suggestion
            if (!this.activeCandidateEvents?.[type][candidates?.[i]?.cid]) {
```

Reply:
Done